# Healthcare Patient No-Show Analysis

## Business Problem
Missed medical appointments reduce healthcare efficiency, increase operational costs, and negatively impact patient outcomes.

The objective of this project is to analyze patient appointment data to identify patterns associated with missed appointments and generate actionable insights that healthcare organizations can use to reduce no-show rates.

---

### Tools Used
- Python
- Pandas
- NumPy
- Matplotlib
- GitHub


In [1]:
# ==========================================
# Step 1: Import Required Libraries
# ==========================================

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

print("Libraries imported successfully!")

Libraries imported successfully!


In [6]:
# ==========================================
# Step 2: Load Dataset
# ==========================================

df = pd.read_csv("medical_appointment_no_shows.csv")

In [7]:
# ==========================================
# Step 3: Preview Dataset
# ==========================================

df.head()

,PatientId,AppointmentID,Gender,ScheduledDay,AppointmentDay,Age,Neighbourhood,Scholarship,Hipertension,Diabetes,Alcoholism,Handcap,SMS_received,No-show
0,2.987250e+13,5642903,F,2016-04-29T18:38:08Z,2016-04-29T00:00:00Z,62,JARDIM DA PENHA,0,1,0,0,0,0,No
1,5.589978e+14,5642503,M,2016-04-29T16:08:27Z,2016-04-29T00:00:00Z,56,JARDIM DA PENHA,0,0,0,0,0,0,No
2,4.262962e+12,5642549,F,2016-04-29T16:19:04Z,2016-04-29T00:00:00Z,62,MATA DA PRAIA,0,0,0,0,0,0,No
3,8.679512e+11,5642828,F,2016-04-29T17:29:31Z,2016-04-29T00:00:00Z,8,PONTAL DE CAMBURI,0,0,0,0,0,0,No
4,8.841186e+12,5642494,F,2016-04-29T16:07:23Z,2016-04-29T00:00:00Z,56,JARDIM DA PENHA,0,1,1,0,0,0,No


In [8]:
# ==========================================
# Step 4: Dataset Shape
# ==========================================

df.shape

(110527, 14)

In [9]:
# ==========================================
# Step 5: Column Names
# ==========================================

df.columns

Index(['PatientId', 'AppointmentID', 'Gender', 'ScheduledDay',
       'AppointmentDay', 'Age', 'Neighbourhood', 'Scholarship', 'Hipertension',
       'Diabetes', 'Alcoholism', 'Handcap', 'SMS_received', 'No-show'],
      dtype='object')

In [10]:
# ==========================================
# Step 6: Dataset Information
# ==========================================

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 110527 entries, 0 to 110526
Data columns (total 14 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   PatientId       110527 non-null  float64
 1   AppointmentID   110527 non-null  int64  
 2   Gender          110527 non-null  object 
 3   ScheduledDay    110527 non-null  object 
 4   AppointmentDay  110527 non-null  object 
 5   Age             110527 non-null  int64  
 6   Neighbourhood   110527 non-null  object 
 7   Scholarship     110527 non-null  int64  
 8   Hipertension    110527 non-null  int64  
 9   Diabetes        110527 non-null  int64  
 10  Alcoholism      110527 non-null  int64  
 11  Handcap         110527 non-null  int64  
 12  SMS_received    110527 non-null  int64  
 13  No-show         110527 non-null  object 
dtypes: float64(1), int64(8), object(5)
memory usage: 11.8+ MB


In [11]:
# ==========================================
# Step 7: Summary Statistics
# ==========================================

df.describe()

,PatientId,AppointmentID,Age,Scholarship,Hipertension,Diabetes,Alcoholism,Handcap,SMS_received
count,1.105270e+05,1.105270e+05,110527.000000,110527.000000,110527.000000,110527.000000,110527.000000,110527.000000,110527.000000
mean,1.474963e+14,5.675305e+06,37.088874,0.098266,0.197246,0.071865,0.030400,0.022248,0.321026
std,2.560949e+14,7.129575e+04,23.110205,0.297675,0.397921,0.258265,0.171686,0.161543,0.466873
min,3.921784e+04,5.030230e+06,-1.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
25%,4.172614e+12,5.640286e+06,18.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
50%,3.173184e+13,5.680573e+06,37.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
75%,9.439172e+13,5.725524e+06,55.000000,0.000000,0.000000,0.000000,0.000000,0.000000,1.000000
max,9.999816e+14,5.790484e+06,115.000000,1.000000,1.000000,1.000000,1.000000,4.000000,1.000000


In [12]:
# ==========================================
# Step 8: Missing Values
# ==========================================

df.isnull().sum()

,0
PatientId,0
AppointmentID,0
Gender,0
ScheduledDay,0
AppointmentDay,0
Age,0
Neighbourhood,0
Scholarship,0
Hipertension,0
Diabetes,0


In [13]:
# ==========================================
# Step 9: Duplicate Records
# ==========================================

df.duplicated().sum()

np.int64(0)

# Initial Observations

## Dataset Overview

- Total Records: **110,527**
- Total Features: **14**
- Missing Values: **None**
- Duplicate Records: **None**

---

## Data Types

### Numerical Columns
- PatientId
- AppointmentID
- Age
- Scholarship
- Hipertension
- Diabetes
- Alcoholism
- Handcap
- SMS_received

### Categorical Columns
- Gender
- Neighbourhood
- No-show

### Date Columns
- ScheduledDay
- AppointmentDay

---

## Initial Findings

- The dataset contains **110,527 patient appointment records**, making it suitable for exploratory data analysis.
- No missing values were detected, indicating good data completeness.
- No duplicate records were found.
- ScheduledDay and AppointmentDay are currently stored as text (object) and will require conversion to datetime format.
- The Age column contains a minimum value of **-1**, indicating the presence of invalid data that must be cleaned.
- The maximum recorded age is **115 years**, which should be verified during data cleaning.
- Further preprocessing is required before visualization and predictive analysis.

---

## Business Insight

The dataset appears to be well-structured with excellent data quality. The primary focus during data cleaning will be handling invalid age values, converting date columns, and preparing variables for exploratory analysis to understand the factors influencing patient no-show behavior.

In [14]:
# Number of unique patients

df["PatientId"].nunique()

62299

In [15]:
# Number of unique neighbourhoods

df["Neighbourhood"].nunique()

81

In [16]:
# Distribution of No-show

df["No-show"].value_counts()

,count
No-show,
No,88208
Yes,22319


In [17]:
# Percentage of No-show

df["No-show"].value_counts(normalize=True) * 100

,proportion
No-show,
No,79.806744
Yes,20.193256


# Business Insights

### Patient Distribution
- The dataset contains **110,527 appointments** from **62,299 unique patients**, indicating that many patients scheduled multiple appointments.

### Geographic Coverage
- Appointments were recorded across **81 neighbourhoods**, allowing geographic analysis of no-show behavior.

### Data Quality
- No missing values or duplicate records were identified, indicating high-quality data suitable for analysis.

### Appointment Outcome
- Approximately **79.8%** of appointments were attended.
- Approximately **20.2%** of appointments resulted in patient no-shows.

### Initial Risk Factors
Further analysis will investigate whether patient no-show rates are influenced by:
- Age
- Gender
- SMS reminders
- Medical conditions
- Appointment waiting time
- Neighbourhood